CMakeLists.txt

```diff
@@ -0,0 +1,169 @@
+add_subdirectory(profiling/chrome-tracing)
+add_subdirectory(profiling/simple-kernel-timer-json)
+add_subdirectory(profiling/simple-kernel-timer)
+add_subdirectory(profiling/space-time-stack)
+
+# External lib connectors
+if(PAPI_FOUND)
+  add_subdirectory(profiling/papi-connector)
+else()
+  message(WARNING "PAPI library not found - skipping papi-connector")
+endif()
+
+#add_subdirectory(profiling/systemtap-connector)
+add_subdirectory(profiling/variorum-connector)
+
+# Build external
+# TODO: tpls/apex
+
+# GPU profilers
+if(KOKKOS_ENABLE_CUDA)
+  #add_subdirectory(profiling/nvprof-connector)
+  #add_subdirectory(profiling/nvprof-focused-connector)
+endif()
+if(KOKKOS_ENABLE_HIP)
+  #add_subdirectory(profiling/roctx-connector)
+endif()
+
+if(DEFINED ENV{VTUNE_HOME})
+  set(VTune_ROOT $ENV{VTUNE_HOME})
+endif()
+if(VTune_ROOT)
+  #add_subdirectory(profiling/vtune-connector)
+  #add_subdirectory(profiling/vtune-focused-connector)
+else()
+  message(WARNING "Set VTUNE_HOME in environment or VTune_ROOT in build options to build VTune connectors")
+  set(VTune_ROOT "" CACHE STRING "Path to VTune Intel compiler")
+endif()
+
+# Note: Caliper docs https://software.llnl.gov/Caliper/build.html
+#       might miss some options, check the top CMakeLists.txt:
+#       https://github.com/LLNL/Caliper/blob/master/CMakeLists.txt
+macro(configure_caliper)
+  #set(BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS}) # Build shared (ON - default) or static library (OFF).
+  #set(BUILD_DOCS OFF)   # Build documentation.
+  #set(BUILD_TESTING OFF) # Build unit tests.
+  #set(WITH_FORTRAN OFF) # Build and install Fortran wrappers.
+  set(WITH_KOKKOS ON) # Enable Kokkos profiling support
+
+  # Build with support for the OpenMP tools interface.
+  if(OpenMP_FOUND)
+    set(WITH_OMPT ON)
+  else()
+    set(WITH_OMPT OFF)
+  endif()
+
+  set(WITH_SAMPLER ON)  # Enable time-based sampling on Linux.
+  set(WITH_TOOLS ON)    # Build Caliper’s tools (i.e, cali-query and mpi-caliquery). Default: On.
+  set(WITH_MPI ${KOKKOS_ENABLE_MPI}) # Build with MPI support.
+  set(WITH_ROCM ${Kokkos_ENABLE_HIP}) # Enable AMD ROCtracer/RocTX support
+  set(WITH_NVTX ${Kokkos_ENABLE_CUDA}) # Build adapters to forward Caliper annotations to NVidia’s nvtx annotation API. Set CUDA_TOOLKIT_ROOT_DIR to the CUDA installation.
+  set(WITH_CUPTI ${Kokkos_ENABLE_CUDA}) # Enable support for CUDA performance analysis
+                                        # (wrapping of driver/runtime API calls and CUDA activity tracing).
+  # Enable PAPI hardware counter service (requires papi)
+  if(PAPI_FOUND)
+    set(WITH_PAPI ON)
+    set(PAPI_PREFIX ${PAPI_ROOT})
+  endif()
+  # Build adapters to forward Caliper annotations to Intel’s VTune annotation API.
+  if(VTune_ROOT)
+    set(WITH_VTUNE ON)
+    set(ITT_PREFIX ${VTune_ROOT}/sdk)
+    # TODO: Caliper doesn't seek ITT in ${VTune_ROOT}/sdk/lib64 or ${VTune_ROOT}/sdk/lib32 (only in .../lib)
+  else()
+    set(WITH_VTUNE OFF)
+  endif()
```

Comment:
Again, you are forcing a configuration which the user may not want. Let them specify it themselves or, at the very least, set these as cache variables so there are default settings but the user can change them.

Reply:
I've changed all settings to cache variables and also minimized number of option defaults imposed on Caliper and Apex by the tools: it nails down to enabling support for Kokkos, MPI, Variorum, VTune, PAPI and CUDA/HIP support - when those are requested/detected (also the compiler if user enables `KokkosTools_REUSE_KOKKOS_COMPILER`).